frontend/src/lib/schemas/slugSchema.ts

```diff
@@ -1,12 +1,24 @@
 import slugify from "@sindresorhus/slugify";
 import { z } from "zod";
 
-export const slugSchema = z
-  .string()
-  .trim()
-  .min(1)
-  .max(32)
-  .refine((val) => val.toLowerCase() === val, "Must be lowercase")
-  .refine((v) => slugify(v) === v, {
-    message: "Invalid slug format"
-  });
+interface SlugSchemaInputs {
+  min?: number;
+  max?: number;
+  field?: string;
+}
+
+export const slugSchema = ({ min = 1, max = 32, field = "Slug" }: SlugSchemaInputs = {}) => {
+  return z
+    .string()
+    .trim()
+    .toLowerCase()
+    .min(min, {
+      message: `${field} field must be at least ${min} character${min === 1 ? "" : "s"}`
+    })
+    .max(max, {
+      message: `${field} field must be at most ${max} character${max === 1 ? "" : "s"}`
+    })
+    .refine((v) => slugify(v, { lowercase: true, separator: "-" }) === v, {
+      message: `${field} field can only contain letters, numbers, and hyphens`
```

Comment:
since it can only be lower case, maybe you want to mention that too? Also by default separator is `-` so we don't need to define?

Reply:
Thought a bit about this;
RE Lowercase
yes, makes sense to include in the message - we convert to LC anyway, and the field might be named differently (where user would assume caps is ok)
Might need to review codebase and unify on calling input labels `slug` whenever the schema is used (instead of Alias etc.)

On default separator - good point
I thought it would accept other characters as a separator, but dosnt seem to be the case - my bad